spec/resolver_spec.rb

```diff
@@ -119,6 +119,52 @@ def initialize(fixture_path)
         resolved = @resolver.resolve([dep], DependencyGraph.new)
         expect(resolved.map(&:payload).map(&:to_s)).to eq(['actionpack (1.2.3)'])
       end
+
+      class SwapChildWithSuccessorsIndex < TestIndex
+
+        # The bug we want to write a regression test for only occurs when
+        # Molinillo processes dependencies in a specific order for the given
+        # index and demands. This sorting logic ensures we hit the repro case
+        # when using the index file "swap_child_with_successors"
+        def sort_dependencies(dependencies, activated, conflicts)
```

Comment:
Not sure how best to handle this. The sorting algorithm in `TestIndex` changes the order and sidesteps the bug, so I copied the sorting we're using in Solve to make the repro case work.